### PR TITLE
[KYUUBI #1634][FEATURE] Redact secret information from SparkSQLEngine UI

### PR DESCRIPTION
**Pull request by minyk**
### _Why are the changes needed?_

SparkSQLEngine's Kyuubi UI tab shows SQL statements. In some cases, these informations are secret information like S3 access/secret key. Spark SQL UI tab already redact with `spark.redaction.string.regex` configuration. Kyuubi also should redact secret informations with this property.

### _How was this patch tested?_
- [X] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [X] Add screenshots for manual tests if appropriate
with below property and query:
spark-defaults.conf:
```
spark.redaction.string.regex (?i)url|access.*|secret.*|password
```
query:
```
SET fs.s3a.access.key=testkey
```

Screen Shot 2021-12-28 at 9 00 07 PM

- [X] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala [hunk 1]
```diff
@@ -69,13 +70,18 @@ class ExecuteStatement(
...
+
   private def executeStatement(): Unit = withLocalProperties {
     try {
       setState(OperationState.RUNNING)
       info(diagnostics)
       Thread.currentThread().setContextClassLoader(spark.sharedState.jarClassLoader)
       // TODO: Make it configurable
       spark.sparkContext.addSparkListener(operationListener)
+      spark.sparkContext.setJobDescription(
+        redact(spark.sessionState.conf.stringRedactionPattern, statement))
```
Comment (ulysses-you) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala:84]: Can we make redacted statment at:

https://github.com/apache/incubator-kyuubi/blob/cb8721fef130771db330c0dd51844ebb13bca7b3/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/SparkOperation.scala#L98-L100

And we can also define the `redactedStatement` at `SparkOperation`.

**Review by ulysses-you — COMMENTED** [review]
thank you for the contribution @minyk !

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala (hunk 1 above)
Comment (yaooqinn) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala:84]: +1 to set there

File: kyuubi-common/src/main/scala/org/apache/kyuubi/operation/AbstractOperation.scala [hunk 2]
```diff
@@ -59,6 +59,7 @@ abstract class AbstractOperation(opType: OperationType, session: Session)
   def getBackgroundHandle: Future[_] = _backgroundHandle
 
   def statement: String = opType.toString
+  def redactedStatement: String = opType.toString
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/operation/AbstractOperation.scala:62]: statement？

Comment (minyk) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/operation/AbstractOperation.scala:62]: do you mean `redactedstatment`? all lowercase?

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala (hunk 1 above)
Comment (minyk) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala:84]: let me try

File: kyuubi-common/src/main/scala/org/apache/kyuubi/operation/AbstractOperation.scala (hunk 2 above)
Comment (yaooqinn) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/operation/AbstractOperation.scala:62]: def redactedStatement: String = statement

Comment (minyk) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/operation/AbstractOperation.scala:62]: Thanks.

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala (hunk 1 above)
Comment (minyk) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala]: move codes to `SparkOperation`

File: kyuubi-common/src/main/scala/org/apache/kyuubi/operation/AbstractOperation.scala (hunk 2 above)
Comment (minyk) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/operation/AbstractOperation.scala]: `redactedStatement` moves to `SparkOperation`

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/SparkOperation.scala [hunk 3]
```diff
@@ -98,6 +101,8 @@ abstract class SparkOperation(opType: OperationType, session: Session)
   protected def withLocalProperties[T](f: => T): T = {
     try {
       spark.sparkContext.setJobGroup(statementId, statement, forceCancel)
+      spark.sparkContext.setJobDescription(
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/SparkOperation.scala:104]: we can use redactedStatement here?

Comment (minyk) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/SparkOperation.scala]: yes we can. apply and action confirmed.

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/SparkOperation.scala [hunk 4]
```diff
@@ -98,6 +101,7 @@ abstract class SparkOperation(opType: OperationType, session: Session)
   protected def withLocalProperties[T](f: => T): T = {
     try {
       spark.sparkContext.setJobGroup(statementId, statement, forceCancel)
+      spark.sparkContext.setJobDescription(redactedStatement)
```
Comment (ulysses-you) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/SparkOperation.scala:104]: we can combine this two line by:

`spark.sparkContext.setJobGroup(statementId, redactedStatement, forceCancel)`

**Review by ulysses-you — APPROVED** [review]
thank you @minyk  for the updating !

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/SparkOperation.scala (hunk 4 above)
Comment (minyk) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/SparkOperation.scala]: Thanks. it will be combined. ;-)

**Review by yaooqinn — APPROVED** [review]
LGTM

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**